src/test/scala/squid/ir/PrettyPrinterTests.scala

```diff
@@ -0,0 +1,392 @@
+package squid
+package ir
+
+import org.scalatest.FunSuite
+import MacroTesters._
+
+class PrettyPrinterTests extends FunSuite {
+  object b extends ir.SimpleAST
+  
+  val Inter = new ir.PrettyPrinter
+  
+  def same[A](xy: (A, A)) = assert(xy._1 == xy._2)
+  def runSame[A](xy: (b.Rep, A), exp: String) = {
+    same((b.reinterpret(xy._1, Inter)())(0), exp)
+  }
+  
+  test("Constants") ({
+    
+    runSame( shallowAndDeep(b){ 42 }, "42" )
```

Comment:
That's an interesting and unexpected way to use the internal/undocumented `shallowAndDeep` macro! :^)

As a side-note, you can also test the pretty-printer from code already expressed in an IR, by doing one of:
```scala
object Embedding extends squid.ir.SimpleAST
import Embedding.Predef._
object P extends squid.ir.PrettyPrinter

Embedding.reinterpret(ir"42".rep, P)()(0)
(ir"42" reinterpretIn P).rep(0)
```